### PR TITLE
Fix 1497 Fix Issue In Loading Overdue

### PR DESCRIPTION
**Pull request by xsalefter**
I explained the problem of #1497 in [its comment](https://github.com/killbill/killbill/issues/1497#issuecomment-996569780).

### CODE REVIEW

File: jaxrs/src/test/java/org/killbill/billing/jaxrs/resources/TestOverdueResource.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        final OverdueResource resource = newOverdueResource();
+        final Response response = resource.getOverdueConfigJson(any());
+
+        Assert.assertEquals(response.getStatus(), 200);
+    }
+
+
+    /**
+     * XML in test have the same content as used in {@link #testGetOverdueConfigJson()}, but with
+     * 'totalUnpaidInvoiceBalanceEqualsOrExceeds' removed in all condition. Purpose of this to proof that
```
Comment (sbrossie) [inline jaxrs/src/test/java/org/killbill/billing/jaxrs/resources/TestOverdueResource.java:131]: Nit: typo `proof` -> `prove`

File: jaxrs/src/test/java/org/killbill/billing/jaxrs/resources/TestOverdueResource.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    /**
+     * Represent invalid overdue that cannot marshalled into JSON. This XML contains the same value as defined in:
+     * https://github.com/killbill/killbill/issues/1497
+     *
+     * This XML causing NullPointerException when JSON marshalling because it missing
+     * 'timeSinceEarliestUnpaidInvoiceEqualsOrExceeds' element.
+     */
+    @Test(groups = "fast", description = "https://github.com/killbill/killbill/issues/1497")
+    public void testGetOverdueConfigJson_noTimeSinceEarliest() throws Exception {
+        final OverdueConfig overdueConfig = getOverdueConfig("overdue_no-TimeSinceEarliest.xml");
```
Comment (sbrossie) [inline jaxrs/src/test/java/org/killbill/billing/jaxrs/resources/TestOverdueResource.java:83]: Your naming conventions are a bit hard to follow, this is a mix of camel/snake cases and `-`... Can we settle on something a bit simpler (e.g under `beatrix/src/test/resources/org/killbill/billing/beatrix/`)

File: jaxrs/src/test/java/org/killbill/billing/jaxrs/resources/TestOverdueResource.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        Assert.assertEquals(response.getStatus(), 200);
+    }
+
+
+    /**
+     * XML in test contains the same value as defined in: https://github.com/killbill/killbill/issues/1497, except that
+     * 'timeSinceEarliestUnpaidInvoiceEqualsOrExceeds' element is added
+     */
+    @Test(groups = "fast", description = "https://github.com/killbill/killbill/issues/1497")
+    public void testGetOverdueConfigJson_withTimeSinceEarliest() throws Exception {
```
Comment (sbrossie) [inline jaxrs/src/test/java/org/killbill/billing/jaxrs/resources/TestOverdueResource.java:116]: Naming convention: Same thing here. Typically we only do camel case and sometimes add `_` when name becomes too unreadable or when there is a need to add enum names with uppercase in the name (example [here](https://github.com/killbill/killbill/blob/master/beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestIntegrationDryRunInvoice.java#L726)

**Review by sbrossie — COMMENTED** [review]
Left a few small comments, I think we are ready for you to add the fix now.

Also read @reshmabidikar comment [here](https://github.com/killbill/killbill/issues/1497#issuecomment-997929065) if not already done.

**Comment by xsalefter** [comment]
@sbrossie fixed your change request, you could check them out. However, reading a comment from @reshmabidikar [here](https://github.com/killbill/killbill/issues/1497#issuecomment-997929065), I think the fix should simply set `DurationJson` with null `TimeUnit` and zero number [like this](https://github.com/killbill/killbill/pull/1549/files#diff-dd393bfad568157db106506efb6b5409b0134bb1904223f3debc2ea56c8caa81R59). What do you think?

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/json/OverdueConditionJson.java [hunk 4]
```diff
@@ -57,7 +57,9 @@ public OverdueConditionJson(@JsonProperty("timeSinceEarliestUnpaidInvoiceEqualsO
     }
 
     public OverdueConditionJson(final OverdueCondition overdueCondition) {
-        this.timeSinceEarliestUnpaidInvoiceEqualsOrExceeds = new DurationJson(overdueCondition.getTimeSinceEarliestUnpaidInvoiceEqualsOrExceeds());
+        this.timeSinceEarliestUnpaidInvoiceEqualsOrExceeds = overdueCondition.getTimeSinceEarliestUnpaidInvoiceEqualsOrExceeds() == null ?
+                                                             new DurationJson(null, 0) :
+                                                             new DurationJson(overdueCondition.getTimeSinceEarliestUnpaidInvoiceEqualsOrExceeds());
         this.controlTagInclusion = overdueCondition.getInclusionControlTagType();
```
Comment (reshmabidikar) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/json/OverdueConditionJson.java:63]: IMO, `this.timeSinceEarliestUnpaidInvoiceEqualsOrExceeds` should be set to null if  `overdueCondition.getTimeSinceEarliestUnpaidInvoiceEqualsOrExceeds() == null`

Comment (xsalefter) [reply jaxrs/src/main/java/org/killbill/billing/jaxrs/json/OverdueConditionJson.java:63]: You're right, just set it to null works, too.

**Review by sbrossie — APPROVED** [review]

File: jaxrs/src/test/java/org/killbill/billing/jaxrs/resources/TestOverdueResource.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+import org.killbill.billing.util.api.TagUserApi;
+import org.killbill.billing.util.callcontext.TenantContext;
+import org.killbill.clock.Clock;
+import org.killbill.xmlloader.XMLLoader;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.google.common.io.Resources;
+
+import static org.mockito.Mockito.*;
```
Comment (pierre) [inline jaxrs/src/test/java/org/killbill/billing/jaxrs/resources/TestOverdueResource.java:42]: Nit: avoid wildcard imports. FWIW You can trick IntelliJ to never do wildcard imports by setting a high threshold number:

image

**Review by pierre — APPROVED** [review]
